source/extensions/priv/server/ntds_jet.c

```diff
@@ -0,0 +1,432 @@
+		return readStatus;
+	}
+	userAccount->accountDisabled = !!(accountControl & NTDS_ACCOUNT_DISABLED);
+	userAccount->accountLocked = !!(accountControl & NTDS_ACCOUNT_LOCKED);
+	userAccount->noPassword = !!(accountControl & NTDS_ACCOUNT_NO_PASS);
+	userAccount->passExpired = !!(accountControl & NTDS_ACCOUNT_PASS_EXPIRED);
+	userAccount->passNoExpire = !!(accountControl & NTDS_ACCOUNT_PASS_NO_EXPIRE);
+	// Grab the Logon Count here
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->logonCount.columnid, &userAccount->logonCount, sizeof(userAccount->logonCount), &columnSize, 0, NULL);
+	if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	// Grab the various Dates and Times
+	readStatus = read_user_dates(ntdsState, accountColumns, pekDecrypted, userAccount);
+	if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	// Grab the NT Hash
+	readStatus = read_user_nt_hash(ntdsState, accountColumns, pekDecrypted, userAccount);
+	if (readStatus != JET_errSuccess && readStatus != JET_wrnColumnNull){
+		return readStatus;
+	}
+	// Grab the LM Hash
+	readStatus = read_user_lm_hash(ntdsState, accountColumns, pekDecrypted, userAccount);
+	if (readStatus != JET_errSuccess && readStatus != JET_wrnColumnNull){
+		return readStatus;
+	}
+	// Grab the Hash History
+	readStatus = read_user_hash_history(ntdsState, accountColumns, pekDecrypted, userAccount);
+	if (readStatus != JET_errSuccess && readStatus != JET_wrnColumnNull){
+		return readStatus;
+	}
+	return JET_errSuccess;
+}
+
+
+/*!
+* @brief Read the current user record's various datetime records.
+* @param ntdsState Pointer to a jetsState struct which contains all the state data for the Jet Instance.
+* @param accountColumns Pointer to an ntdsState struct which will hold all of our column definitions.
+* @param pekDecrypted Pointer to a decryptedPEK structure that holds our decrypted PEK
+* @param userAccount Pointer to an ntdsAccount struct that will hold all of our User data
+* @returns Indication of sucess or failure.
+*/
+JET_ERR read_user_dates(struct jetState *ntdsState, struct ntdsColumns *accountColumns, struct decryptedPEK *pekDecrypted, struct ntdsAccount *userAccount){
+	JET_ERR readStatus = JET_errSuccess;
+	unsigned long columnSize = 0;
+	FILETIME accountExpiry;
+	SYSTEMTIME accountExpiry2;
+	FILETIME lastLogon;
+	SYSTEMTIME lastLogon2;
+	FILETIME lastPass;
+	SYSTEMTIME lastPass2;
+	// Grab the account expiration date/time here
+	readStatus = JetRetrieveColumn(ntdsState->jetSession, ntdsState->jetTable, accountColumns->accountExpiry.columnid, &accountExpiry, sizeof(accountExpiry), &columnSize, 0, NULL);
+	if (readStatus != JET_errSuccess){
+		return readStatus;
+	}
+	//Convert the FILETIME to a SYSTEMTIME so we can get a human readable date
+	FileTimeToSystemTime(&accountExpiry, &accountExpiry2);
+	int dateResult = GetDateFormat(LOCALE_SYSTEM_DEFAULT, DATE_LONGDATE, &accountExpiry2, NULL, userAccount->expiryDate, 255);

```

Comment:
nope, you're not misunderstanding. those buffers were originally 255, i then cut them down to an appropriate size and never changed these calls. good catch